lib/__tests__/themes.test.ts

```diff
@@ -0,0 +1,84 @@
+import findup from 'findup-sync';
+import { getHubSpotWebsiteOrigin } from '../urls';
+import { getThemeJSONPath, getThemePreviewUrl } from '../cms/themes'; // Adjust the path to your module
+import { getEnv } from '../../config';
+import { ENVIRONMENTS } from '../../constants/environments';
+
+jest.mock('findup-sync');
+jest.mock('../urls');
+jest.mock('../../config');
+jest.mock('../../constants/environments', () => ({
+  ENVIRONMENTS: {
+    PROD: 'https://prod.hubspot.com',
+    QA: 'https://qa.hubspot.com',
+  },
+}));
+
+const mockedFindup = findup as jest.MockedFunction<typeof findup>;
+const mockedGetEnv = getEnv as jest.MockedFunction<typeof getEnv>;
+const mockedGetHubSpotWebsiteOrigin =
+  getHubSpotWebsiteOrigin as jest.MockedFunction<
+    typeof getHubSpotWebsiteOrigin
+  >;
+
+describe('getThemeJSONPath', () => {
+  it('should return the theme.json path if found', () => {
+    mockedFindup.mockReturnValue('/path/to/theme.json');
+
+    const result = getThemeJSONPath('/some/path');
+
+    expect(findup).toHaveBeenCalledWith('theme.json', {
+      cwd: '/some/path',
+      nocase: true,
+    });
+    expect(result).toBe('/path/to/theme.json');
+  });
+
+  it('should return null if theme.json is not found', () => {
+    mockedFindup.mockReturnValue(null);
+
+    const result = getThemeJSONPath('/some/path');
+
+    expect(findup).toHaveBeenCalledWith('theme.json', {
+      cwd: '/some/path',
+      nocase: true,
+    });
+    expect(result).toBeNull();
+  });
+});
+
+describe('getThemePreviewUrl', () => {
+  it('should return the correct theme preview URL for PROD environment', () => {
+    mockedFindup.mockReturnValue('/path/to/theme.json');
+    mockedGetEnv.mockReturnValue('prod');
+    mockedGetHubSpotWebsiteOrigin.mockReturnValue('https://prod.hubspot.com');
+
+    const result = getThemePreviewUrl('/path/to/file', 12345);
+
+    expect(getEnv).toHaveBeenCalledWith(12345);
+    expect(getHubSpotWebsiteOrigin).toHaveBeenCalledWith(ENVIRONMENTS.PROD);
+    expect(result).toBe(
+      'https://prod.hubspot.com/theme-previewer/12345/edit/to'
```

Comment:
I'm not following, do you mean you think the path is incorrect?

Reply:
I did some digging for this and I think I know what's going on.

This getThemePreviewURL util gets called during `hs upload` [here](https://github.com/HubSpot/hubspot-cli/blob/main/packages/cli/commands/upload.js#L58). After all of the uploads finish, the command checks to see if the user ran the upload from within a CMS theme folder. It checks by looking for the theme.json file. 

The theme preview path uses the name of the folder that the theme.json file lives in. So if your folder structure looks like this:
```
/ 
..my-theme/
....theme.json
....templates/
....assets/    
```

The `getThemeNameFromPath` function takes the full absolute path to the theme.json file and then returns the folder that the file lives inside of. So it would generate something like `/Users/brodgers/src/my-theme/theme.json` and would return `my-theme` as the path.

The theme preview path would be something like `https://app.hubspot.com/theme-previewer/<account_id>/edit/my-theme`. So your test is working correctly. The test would be easier to follow if you changed the[ mock of findUp](https://github.com/HubSpot/hubspot-local-dev-lib/pull/195/files#diff-514a5be7f3e52cf7cb8d108d7a7be2630da79c7dbfe7dfb428abd04c3c22dc21R26) to return something like `/src/my-theme/theme.json`

Weirdly, this functionality doesn't work for me when I try to test it while running the CLI locally. But you can test it by running `hs project upload` from the root of a CMS theme (just use the default values for each prompt)

Reply:
Edit: I think I have a fix for `getThemePreviewUrl` to make it work locally. I can PR it